src/Sniffer.php

```diff
@@ -0,0 +1,278 @@
+                    // 3 is the exit code phpcs uses for errors like this
+                    throw new DeepExitException("Can't find file {$dummy->realPath} to set new content", 3);
+                }
+
+                /** @var FencedCode $mdBlock */
+                $mdBlock = $codeBlocks[$dummy->path]['md'];
+                $indent = str_repeat(' ', $mdBlock->getOffset());
+                $origBlockContent = preg_replace('/^/m', $indent, $mdBlock->getLiteral());
+                $newBlockContent = preg_replace('/^/m', $indent, preg_replace('/\s*<\?php\n?/', '', $dummy->getContent()));
+                $newFileContent = str_replace($origBlockContent, $newBlockContent, file_get_contents($dummy->realPath));
+
+                file_put_contents($dummy->realPath, $newFileContent);
+            }
+        }
+
+        $sniffer->reporter->printReports();
+
+        if ($numErrors === 0) {
+            // No errors found.
+            return 0;
+        } else if ($sniffer->reporter->totalFixable === 0) {
+            // Errors found, but none of them can be fixed by PHPCBF.
+            return 1;
+        } else {
+            // Errors found, and some can be fixed by PHPCBF.
+            return 2;
+        }
+    }
+
+    private function prepareConfig(bool $usingExplicitStandard, string $lintLanguage, bool $fixing): Config
+    {
+        // Creating the Config object populates it with all required settings based on the phpcs/phpcbf CLI arguments provided.
+        $config = new Config();
+
+        // We don't support STDIN for passing markdown in
+        if ($config->stdin === true) {
+            // 3 is the exit code phpcs uses for errors like this
+            throw new DeepExitException('STDIN isn\'t supported', 3);
+        }
+
+        // Ensure we can find and lint markdown files
+        $config->extensions = array_merge($config->extensions, ['md' => $lintLanguage]);
+        // We're not passing the sniffer any real files, so caching could be unreliable
+        $config->cache = false;
+        // We must sniff all "files" sequentially - asyncronous sniffing isn't supported
+        $config->parallel = 1;
+
+        // If the user hasn't defined an explicit standard, and there's no default standards file to use,
+        // use our customised PSR12 standard
+        if (!$usingExplicitStandard && $config->standards === ['PEAR']) {
+            $config->standards = [__DIR__ . '/../phpcs.default.xml'];
+        }
+
+        if ($fixing) {
+            // Override some of the command line settings that might break the fixes.
+            $config->generator = null;
+            $config->explain = false;
+            $config->interactive = false;
+            $config->cache = false;
+            $config->showSources = false;
+            $config->recordErrors = false;
+            $config->reportFile = null;
+            $config->reports = [FixerReport::class => null];
+            $config->dieOnUnknownArg = false;
+        }
```

Comment:
See [`Runner::runPHPCBF()`](https://github.com/squizlabs/PHP_CodeSniffer/blob/321d252dc9c180a085ecb6acc8a861aba8c4da2d/src/Runner.php#L190-L202)

Reply:
Should make this an inline comment

Reply:
Done